src/main/java/io/reactivex/Observable.java

```diff
@@ -420,4 +422,129 @@ public static int bufferSize() {
             return v;
         });
     }
+    
+    public final Observable<T> observeOn(Scheduler scheduler) {
+        return observeOn(scheduler, false, bufferSize());
+    }
+    
+    public final Observable<T> observeOn(Scheduler scheduler, boolean delayError) {
+        return observeOn(scheduler, delayError, bufferSize());
+    }
+    
+    public final Observable<T> observeOn(Scheduler scheduler, boolean delayError, int bufferSize) {
+        // TODO implement
+        throw new UnsupportedOperationException();
+    }
+    
+    public final Observable<T> subscribeOn(Scheduler scheduler) {
+        // TODO implement
+        throw new UnsupportedOperationException();
+    }
+    
+    public final Observable<T> unsubscribeOn(Scheduler scheduler) {
+        // TODO implement
+        throw new UnsupportedOperationException();
+    }
+    
+    public final Observable<T> cache() {
+        return CachedObservable.from(this);
+    }
+
+    public final Observable<T> cache(int capacityHint) {
+        return CachedObservable.from(this, capacityHint);
+    }
+
+    public final ConnectableObservable<T> publish() {
+        return publish(bufferSize());
+    }
+
+    public final ConnectableObservable<T> publish(int bufferSize) {
+        if (bufferSize <= 0) {
+            throw new IllegalArgumentException("bufferSize > 0 required but it was " + bufferSize);
+        }
+        return OperatorPublish.create(this, bufferSize);
+    }
+
+    public final <R> Observable<R> publish(Function<? super Observable<T>, ? extends Observable<R>> selector) {
+        return publish(selector, bufferSize());
+    }
+
+    public final <R> Observable<R> publish(Function<? super Observable<T>, ? extends Observable<R>> selector, int bufferSize) {
+        if (bufferSize <= 0) {
+            throw new IllegalArgumentException("bufferSize > 0 required but it was " + bufferSize);
+        }
+        return OperatorPublish.create(this, selector, bufferSize);
+    }
+
+    public final ConnectableObservable<T> replay() {
+        return OperatorReplay.createFrom(this);
+    }
+
+    public final <R> Observable<R> replay(Function<? super Observable<T>, ? extends Observable<R>> selector) {
+        return OperatorReplay.multicastSelector(this::replay, selector);
+    }
+    
+    public final <R> Observable<R> replay(Function<? super Observable<T>, ? extends Observable<R>> selector, final int bufferSize) {
+        return OperatorReplay.multicastSelector(() -> replay(bufferSize), selector);
+    }
+    
+    public final <R> Observable<R> replay(Function<? super Observable<T>, ? extends Observable<R>> selector, int bufferSize, long time, TimeUnit unit) {
```

Comment:
See #1353 for concern about the parameter name "time" (similar parameters in other operators are called "timespan", "timeout", "interval", "period", "intervalDuration", etc.; those parameters that have pretty much the same function should have the same name).